ecs-update/ecs-update.go

```diff
@@ -1798,13 +2026,100 @@ func fieldsYMLUseExternalECS(f *ast.File, ff fleetpkg.FieldsFile) (changed bool,
 		replacement := newNode(fmt.Sprintf("name: %s\nexternal: ecs", quoteKey(origField.Name)))
 
 		if err = p.ReplaceWithNode(f, replacement); err != nil {
-			return false, fmt.Errorf("faield to replace node: %w", err)
+			return false, fmt.Errorf("failed to replace node: %w", err)
 		}
 		changed = true
 	}
 	return changed, nil
 }
 
+// fieldsYMLDropExternalECS will find fields that can use 'external: ecs' and remove
+// their definitions. It will use the latest version of ECS to determine if the field
+// exists in ECS.
+func fieldsYMLDropExternalECS(f *ast.File, ff fleetpkg.FieldsFile) (changed bool, err error) {
+	// Use the flat representation so that we have access to the complete field name.
+	flatFields, err := fleetpkg.FlattenFields(ff.Fields)
+	if err != nil {
+		return false, err
+	}
+
+	for _, field := range flatFields {
+		// Does the field exist in ECS?
+		ecsField, _ := ecs.Lookup(field.Name, "")
+		if ecsField == nil {
+			continue
+		}
+
+		if !couldBeECS(field, ecsField) {
```

Comment:
I was testing this and it appears that there are four fields that must be kept no matter what. I'm not sure where this check comes from.

```
Error: building package failed: invalid content found in built zip package: found 4 validation errors:
   1. expected field "data_stream.dataset" with type "constant_keyword" not found in datastream "audit"
   2. expected field "data_stream.namespace" with type "constant_keyword" not found in datastream "audit"
   3. expected field "@timestamp" with type "date" not found in datastream "audit"
   4. expected field "data_stream.type" with type "constant_keyword" not found in datastream "audit"
```